scripts/ansible-common.sh

```diff
@@ -0,0 +1,120 @@
+  echo $ANSIBLE_EXTRA_VARS | jq -M . > /tmp/extra_ansible_args.json
+  ANSIBLE_EXTRA_ARGS="${ANSIBLE_EXTRA_ARGS} -e @/tmp/extra_ansible_args.json"
+fi
+if [ -n "$TAGS" ]; then
+  ANSIBLE_EXTRA_ARGS="${ANSIBLE_EXTRA_ARGS} --tags $(echo $TAGS | jq -r '. | join(",")')"
+fi
+if [ -n "$SKIP_TAGS" ]; then
+  ANSIBLE_EXTRA_ARGS="${ANSIBLE_EXTRA_ARGS} --skip-tags $(echo $SKIP_TAGS | jq -r '. | join(",")')"
+fi
+if [ "$AWS_INVENTORY" == "auto" ] && [ -n "$AWS_ACCESS_KEY_ID" ] || [ "${AWS_INVENTORY,,}" == "true" ]; then
+  # Render ec2.ini template from envvars
+  envsubst < /etc/ansible/hosts-template/ec2.ini.template > /etc/ansible/hosts/ec2.ini
+  cp /etc/ansible/hosts-template/ec2.py /etc/ansible/hosts/
+  ANSIBLE_EXTRA_ARGS="${ANSIBLE_EXTRA_ARGS} -i /etc/ansible/hosts/ec2.py"
+fi
+
+if [ -z "${ANSIBLE_PLUGIN_AZURE_HOST}" ]; then
+  if [ "${AZURE_USE_PRIVATE_IP,,}" == "true" ]; then
+      export ANSIBLE_PLUGIN_AZURE_HOST="${DEFAULT_ANSIBLE_PLUGIN_AZURE_HOST_PRIVATE}"
+  else
+      export ANSIBLE_PLUGIN_AZURE_HOST="${DEFAULT_ANSIBLE_PLUGIN_AZURE_HOST}"
+  fi
+fi
+if [ "$AZURE_INVENTORY" == "auto" ] && [ -n "$AZURE_SUBSCRIPTION_ID" ] || [ "${AZURE_INVENTORY,,}" == "true" ]; then
+  if (( $(echo "${ANSIBLE_VERSION%.*} >= 2.8" |bc -l) )); then
+    # Render default.azure_rm.yml template from envvars
+    envsubst < /etc/ansible/hosts-template/default.azure_rm.yml.template > /etc/ansible/hosts/default.azure_rm.yml
+    ANSIBLE_EXTRA_ARGS="${ANSIBLE_EXTRA_ARGS} -i /etc/ansible/hosts/default.azure_rm.yml"
+  else
+    cp /etc/ansible/hosts-template/azure_rm.py /etc/ansible/hosts/
+    ANSIBLE_EXTRA_ARGS="${ANSIBLE_EXTRA_ARGS} -i /etc/ansible/hosts/azure_rm.py"
+  fi
+fi
+
+# Setup SSH access
+eval $(ssh-agent -s)
+
+# SSH keys
+KEY_ID=1
+if [ -n "$SSH_PRIVATE_KEYS" ]; then
+	IFS=$'\n'
+	for key in $(python -c "import json, os; print('\n'.join([ '%s' % v for v in json.loads(os.environ['SSH_PRIVATE_KEYS'].replace('\\n','\\\\\\\n'))]))"); do
+		# Use the first key as default SSH_PRIVATE_KEY if not defined
+		if [ -z "$SSH_PRIVATE_KEY" ]; then
+			SSH_PRIVATE_KEY=$(echo -e $key)
+			continue
+		fi
+		echo -e "$key" > /root/.ssh/id_rsa${KEY_ID}
+		chmod 600 /root/.ssh/id_rsa${KEY_ID}
+		ssh-add /root/.ssh/id_rsa${KEY_ID}
+		KEY_ID=$((KEY_ID+1))
+	done
+	unset IFS
+fi
+
+if [ -n "$SSH_PRIVATE_KEY" ]; then
+  # Root ssh key
+  echo "${SSH_PRIVATE_KEY}" > /root/.ssh/id_rsa
+  chmod 600  /root/.ssh/id_rsa
+  ssh-add /root/.ssh/id_rsa
+fi
+set -x
+if [ -n "$SSH_JUMP_URL" ]; then
+  export ANSIBLE_SSH_ARGS="$ANSIBLE_SSH_ARGS -o 'ProxyJump=$SSH_JUMP_URL' -o 'ForwardAgent=yes'"
+
+# DEPRECATED jump parameter for a bastion server. Use SSH_JUMP_URL instead
+elif [ -n "$BASTION_URL" ]; then
+  export ANSIBLE_SSH_ARGS="$ANSIBLE_SSH_ARGS"' -o ProxyCommand="ssh -W %h:%p -q '${BASTION_URL}'"'
+  #echo "ansible_ssh_common_args: '-o ProxyCommand=\"ssh -W %h:%p -q ${BASTION_URL}\"'" >> $ANSIBLE_PLAYBOOK_PATH/group_vars/all
+fi
+
+if [ -n "${ANSIBLE_LIMIT_HOSTS}" ]; then
```

Comment:
Since this ansible-cli use `ANSIBLE_TARGET_PATTERN` this condition should be in the `ansible-runner`

Reply:
nevermind, we can use `--limit` with ad-hoc ansible